### ISSUE TITLE
Micronaut JacocoReportAggregationPlugin broken in Gradle 8.2

### ISSUE THREAD

**Issue by tresat** [gradle/gradle#25618]
### Expected Behavior

Success with Gradle 8.1.1: https://ge.micronaut.io/s/z4db5b7satq26/dependencies?focusedDependency=WzAsMCwxMjM0LFswLDAsWzEyMzRdXV0&toggled=W1swXSxbMCwwXV0

### Current Behavior

Failure with Gradle 8.2: https://ge.micronaut.io/s/kruw4j3titqcs/failure

### Context (optional)

Selection of `classDirectories` in the `JacocoReport` task is failing due to too many matching variants when a project applies the `jacoco-report-aggregation` plugin and its project dependency applies the `java-test-fixtures` plugin and contains test fixtures, which also make a `classes` variant such as:

```
- Configuration ':micronaut-openapi-generator:testFixturesRuntimeElements' variant classes declares a component, preferably in the form of class files:
```

available.  The failure is due to an `ArtifactView` that uses `withVariantReselection()`, the originating `Configuration` reported in the error message is not problematic.  Reselection should **not** be used to select class directories here.

### Steps to Reproduce

See new integration test in `JacocoAggregationIntegrationTest` or rerun the Micronaut build at [this commit](https://github.com/micronaut-projects/micronaut-openapi/commit/7ad6aa1a49a8e90a05b524ff16d54173ba078f12) using Gradle 8.2.

### Gradle version

8.2

### Gradle version that used to work

8.1.1

**Comment by asoftwareguy** [comment]
We are having the same issue in our project using `jacocoAggregation`:

`Could not determine the dependencies of task ':code-coverage-report:check'.
> Could not create task ':code-coverage-report:testCodeCoverageReport'.
   > Could not resolve all files for configuration ':code-coverage-report:aggregateCodeCoverageReportResults'.
      > The consumer was configured to find a component, preferably in the form of class files. However we cannot choose between the following variants of project :app:
          - Configuration ':app:apiElements' variant classes declares a component, preferably in the form of class files:
              - Unmatched attributes:
                  - Provides attribute 'artifactType' with value 'java-classes-directory' but the consumer didn't ask for it
                  - Provides a library but the consumer didn't ask for it
                  - Provides its dependencies declared externally but the consumer didn't ask for it
                  - Provides compatibility with Java 17 but the consumer didn't ask for it
                  - Provides compile-time but the consumer didn't ask for it
          - Configuration ':app:testFixturesRuntimeElements' declares a component, packaged as a jar:
              - Unmatched attributes:
                  - Provides attribute 'artifactType' with value 'jar' but the consumer didn't ask for it
                  - Provides a library but the consumer didn't ask for it
                  - Provides its dependencies declared externally but the consumer didn't ask for it
                  - Provides compatibility with Java 17 but the consumer didn't ask for it
                  - Provides runtime but the consumer didn't ask for it
           ...`

**Comment by tresat** [comment]
@asoftwareguy - Looks like the same error.  This fix should be available in Gradle 8.2.1, coming very soon.

**Comment by ljacomet** [comment]
* Fixed with #25625

**Comment by marcphilipp** [comment]
FWIW this change broke Jacoco aggregation in JUnit's build which uses a custom variant to use the shadow-transformed classes which was no longer resolved after removing `withVariantReselection()`.

Workaround: https://github.com/junit-team/junit5/commit/45ad2b2a2175e0acf636e27c5f5cd0315e463caf
